[Bryan Kuan] iP

bkkuan opened the pull request:
**### Let Bryan.Me Handle Your Tasks**

"Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort." — Paul J. Meyer [Source Here](https://www.brainyquote.com/quotes/paul_j_meyer_160829)

Bryan.Me helps you track tasks effortlessly. It’s:

-   Simple

-   Lightning-fast ⚡


🚀 **Getting Started**

1.  Download the latest release [here](https://github.com/bkkuan/ip/releases).

2.  Run java -jar Bryan.jar in your terminal.

3.  Start managing tasks with commands like todo Buy milk.

Key Features

- [ ] Core Functionality

- [ ] Mark tasks as done

- [x] Save data automatically

- [x] Deadline support (Level 8 done!)

- [ ] Reminders

- [ ] Weekly reports

Here’s a small part of the code that shows the main method:

    public static void main(String[] args) {
        new Bryan("data/bryan.txt").run();
    }

Need Help?

Visit the [User Guide](https://github.com/bkkuan/ip/wiki) or raise an issue!
Built with ❤️ in Java.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level-0: Rename main class, Add greeting, Exit

Commit:
Implement Level-1: Echo functionality

Commit:
Level-2: Add support for storing and listing tasks

Commit:
Level-3: Add support for marking and unmarking tasks

Commit:
Updated Level-4 with unique task handling logic

Commit:
Completed set up for Automated Text UI Testing

Commit:
Updated exceptions to handle errors

Commit:
Added the delete functionality to delete tasks

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+            }
+            task = new Todo(information);
+        }
+        else if (input.startsWith("deadline ")) {
```
It looks like there may be a small formatting issue here, shouldn't this be formatted as `} else if` to follow the project's coding style instead? 

I noticed the same issue in several other places too.

ToxOptimism commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,61 @@
+                    int taskNumber = Integer.parseInt(input.split(" ")[1]) - 1;
+                    taskManager.taskscompleted(taskNumber);
+                } catch (Exception e) {
+                    System.out.println("Error, use mark <task number>'.");
```
Perhaps you could do `System.out.println(e.getMessage());` here. This could allow you to use a single try-catch across the entire if condition instead.

ToxOptimism commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,61 @@
+                    System.out.println("Error, use delete <task number>.");
+                }
+            }
+            else {
```
It looks like there may be a small formatting issue here, shouldn't this be formatted as `} else` to follow the project's coding style instead?

I noticed the same issue in several other places too.

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+        }
+    }
+    //mark tasks done
+    public void taskscompleted(int taskNumber) {
```
Perhaps the method could be called `markTask` here such that it is camelCase and counts as a verb to follow coding standards?

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+        }
+    }
+    //unmark tasks
+    public void tasksnotcompleted(int taskNumber) {
```
Perhaps the method could be called `unmarkTask` here such that it is camelCase and counts as a verb to follow coding standards?

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+        System.out.println("  " + task);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+    //print tasks
```
Since the two different methods can be considered different logical units, perhaps a a line break should be added here to help improve code readability? There are also other lines which could be similarly improved.

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+
+public class TaskManager {
+    private ArrayList<Tasks> tasks;
+    public TaskManager() {
```
Since attributes and constructors are different logical units, perhaps a a line break should be added here to help improve code readability? There are also other lines which could be similarly improved.

ToxOptimism commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,61 @@
+        sc.close();
+    }
+
+
```
It looks like there may be a small formatting issue here, shouldn't there only be one line break here?

ToxOptimism commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,12 @@
+public class Deadline extends Tasks {
+    protected String by;
+    public Deadline(String information, String by) {
```
Since attributes and constructors are different logical units, perhaps a a line break should be added here to help improve code readability? There are also other lines which could be similarly improved.

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+        else if (input.startsWith("event ")) {
+            String details = input.substring(6).trim();
+            String[] parts = details.split(" /from ", 2);
+            if (parts.length < 2 || !parts[1].contains(" /to ")) {
```
Perhaps the logical unit of identifying invalid usage could be broken up by a new line to help improve code readability? I have noticed other areas which face the same issue.

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+        tasks = new ArrayList<>();
+    }
+
+    //add task
```
Perhaps consider using JavaDocs for your method and class comments?

JavaDocs comments can help to improve readability.

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+            task = new Todo(information);
+        }
+        else if (input.startsWith("deadline ")) {
+            String details = input.substring(9).trim();
```
Perhaps use `detail` instead to indicate that the parameter here is not a collection of objects?

ToxOptimism commented on src/main/java/Tasks.java:
```diff
@@ -0,0 +1,26 @@
+public class Tasks {
+    protected String information;
```
Is there any reason why you used the variable name `information` here?

It appears that some of your exception descriptions instead use the term `description`

ToxOptimism commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,96 @@
+import java.util.ArrayList;
+
+public class TaskManager {
```
I like that you used a TaskManager class to abstract the functionality of the task list.

ToxOptimism reviewed (commented):
Overall I found your code is organized well though there are a few areas where small tweaks to coding standards and style could make the code more consistent and easier to read.

Commit:
Implemented auto save functionality with file storage

Commit:
Merge branch-Level-7: Auto save feature

Commit:
Update code to be more concise and easier to manage in saving and loading data

Commit:
Merge branch-Level-7-update: Updated changes

Commit:
Added data handling for deadlines using LocalDate

Commit:
Merge branch-Level-8: Data handling for deadlines

Commit:
refactor: Implement OOP structure with Ui, Parser, and Command classes

Commit:
Merge A-MoreOOP: OOP refactoring

Commit:
Merge A-Gradle: Add Gradle support

bhavinaa commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,39 @@
+import bryan.taskmanager.TaskManager;
+import bryan.ui.Ui;
+
+;
```
not really sure what this is doing over here

bhavinaa commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,39 @@
+import bryan.command.Command;
```
if i am not wrong, by the Java naming conventions suggestions, the package names should start with a lower case

bhavinaa commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,39 @@
+            } catch (BryanException e) {
+                ui.showError(e.getMessage());
+            }
+        }
```
this only catches one type of exception, the bryanException. You could consider catching the other generic exceptions like a fallback

bhavinaa commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,39 @@
+
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
you could consider abstracting this out, into process commands for example for better readability

bhavinaa commented on src/main/java/Bryan.java:
```diff
@@ -0,0 +1,39 @@
+    private final TaskManager taskManager;
+    private final Ui ui;
+
+    public Bryan(String filePath) {
```
this might make unit testing harder later, as each instance of bryan makes its own instance of ui, storage, and taskManager. It might help if there is a way to mock the dependency, by having a constructor which allows you to pass in the relevant objects as well

bhavinaa commented on src/main/java/bryan/command/AddCommand.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    @Override
+    public void execute(TaskManager taskManager, Ui ui, Storage storage) throws BryanException {
```
the naming of execute is a bit general. it is unclear as to what is the purpose of the method till we read into it. It might help if there is a more descriptive name. But I do understand it is a bit hard to standardise as it is a overridden method

bhavinaa commented on src/main/java/bryan/command/AddCommand.java:
```diff
@@ -0,0 +1,26 @@
+    @Override
+    public void execute(TaskManager taskManager, Ui ui, Storage storage) throws BryanException {
+        taskManager.addTask(input);
+        storage.save(taskManager.getTasks());
```
It might be easier to read and understand if you help to abstract this out. The taskManager.getTasks().size() - 1 is a bit dense and can be extracted into a variable for clarity.

bhavinaa reviewed (commented):
This is my Second PR. You have a very good code base and structure. I have tried to give some inputs at areas i think is appropriate.

Commit:
test: Added JUnit tests

Commit:
Merge A-JUnit: Unit tests

Commit:
added JavaDoc comments to classes and methods

Commit:
Updated all codes to be in proper coding standards

Commit:
Merge branch-A-CodingStandard: Proper Coding Standard

Commit:
Added find command to search tasks by keyword

Commit:
Updated code to follow git standard

Commit:
Initial commit: Implement Bryan JavaFX application with event handing and custom dialog boxes

Commit:
Initial commit: Implement Bryan JavaFX application with FXML and custom dialog boxes

Commit:
Level 10: Integrate full CLI commands (todo, deadline, event, mark, unmark, find, delete, list, bye) into GUI

Commit:
Merge branch-Level-10 with Integration of CLI commands into GUI

Commit:
Add assertions to document critical assumptions

Add assert statements in TaskManager.validateIndex() to ensure that the
provided index is within the bounds of the task list, and in
TaskManager.createTask() to confirm that the input is not null or empty.
Also, add an assertion in Deadline.setBy() to verify that the new due
date is not null.

Additionally, update Parser.parse() to assert that the input is neither
null nor empty. These assertions help catch programming errors early and
serve as documentation of important assumptions.

Commit:
Improve code quality in task status updates

Refactor TaskManager.markTask() and unmarkTask() by extracting a helper
method (formatStatusChangeMessage) to format status change messages.
This reduces code duplication and improves readability in the task
management logic.

This change follows code quality principles by promoting code reuse and
clear separation of formatting logic.

Commit:
Add assertions for validation

Commit:
Add assertion in addTask method to ensure tasks is not null

bkkuan committed:
Merge A-Assertions: Use assertions to validate input and prevent errors

Add assertions to validate input and prevent errors

Commit:
Fix missing assert for index bounds check

bkkuan committed:
Merge pull request #3 from bkkuan/branch-A-Assertions

Fix missing assert for index bounds check

Commit:
refactor(task-manager): extract formatStatusChangeMessage helper for
status updates

TaskManager.markTask() and unmarkTask() contain duplicated code for
formatting status change messages, making the task management logic
harder to read. This duplication increases maintenance effort and risks
inconsistency.

Refactor TaskManager.markTask() and unmarkTask() by extracting a helper
method, formatStatusChangeMessage, to handle status message formatting.
This reduces code duplication, improves readability, and aligns with
code quality principles like reusability and separation of concerns.

This change targets the task status update logic specifically and does
not alter the core task marking functionality.

bkkuan committed:
Merge PR #3: Refactor TaskManager for improved code quality

This merge integrates changes from bkkuan/branch-A-Assertions (PR #3) to enhance
code quality in the TaskManager class. The original markTask() and unmarkTask()
methods contained duplicated logic for formatting status change messages, which
reduced readability and increased maintenance effort.

The refactoring extracts this logic into a new helper method,
formatStatusChangeMessage, used by both markTask() and unmarkTask(). This
eliminates duplication, improves readability, and adheres to code quality
principles like reusability and separation of concerns.

No functional changes are introduced—task marking behavior remains unchanged.
The changes were tested manually by marking and unmarking tasks to ensure status
messages remain consistent.

Commit:
Resolve merge conflicts in Parser and TaskManager

Commit:
Add snooze feature to postpone deadline tasks

Users cannot reschedule deadlines, limiting flexibility in task
management.

Add SnoozeCommand to parse "snooze [task number] [yyyy-mm-dd]" format.
Update TaskManager to modify deadline due dates and confirm changes.
Implement setter in Deadline class to update due dates. Extend Parser to
recognize "snooze" command.

Enable this feature to improve usability by allowing task postponement
with a simple, date-specific command structure consistent with existing
syntax.

This change enhances functionality without altering core task handling
logic.

Commit:
Merge branch Snooze into master - Add task snoozing functionality

Commit:
Set up product website and add user guide screenshot

Commit:
Merge branch-UserGuide: Add Bryan User Guide and screenshot

Commit:
chore: remove broken submodule reference for javafx-tutorial

Commit:
fix: remove broken submodule reference

Commit:
chore(release): publish version 0.2

Commit:
fix: update build.gradle to support cross-platform JAR
